### ISSUE TITLE
Rootless privileged podman silently fails to mount some devices

### ISSUE THREAD

**Issue by mtalexan** [containers/podman#16701]
/kind bug

**Description**
Rootless podman `run` with `--privileged --security-opt label=disable` silently fails to mount/create device nodes inside the container when using the `--device` option even if the user has permissions to them granted thru group membership.  

It seems that for some reason podman only checks if the device node is directly owned by the calling user or if the device node is available to everyone (o=rw).  It does not check if the device node is accessible thru group membership.  Additionally, when it hits this case it fails silently, producing no warning or error output about the failure.

**Steps to reproduce the issue:**

1. Identify or create a device node without o=rw permissions, not owned by the user, but with a group the user is a member of (let's call it `/dev/testdev` for the sake of this example)
2. `podman run --privileged --security-opt label=disable -it --device /dev/testdev bash`
3. Inside the container now: `ls -la /dev/`

**Describe the results you received:**
`/dev/testdev` is not present in the container

**Describe the results you expected:**
`/dev/testdev` is present in the container, or an error message was produced during the `podman run` command.

**Additional information you deem important (e.g. issue happens only occasionally):**

This is *not* an [issue of device node permissions in the container](https://github.com/containers/podman/blob/main/troubleshooting.md#20-passed-in-devices-or-files-cant-be-accessed-in-rootless-container) (or [this one](https://github.com/containers/podman/blob/main/troubleshooting.md#35-passed-in-devices-or-files-cant-be-accessed-in-rootless-container-uidgid-mapping-problem)), the issue is that the device node doesn't appear in the container at all.  

In particular this affects Debian-like distros when trying to use the `/dev/kvm` device, which has permissions like:
```
$ ls -la /dev/kvm
crw-rw----+ 1 root kvm 10, 232 Nov 30 17:38 /dev/kvm
```
This is notably different (for unknown reasons) from the Fedora-like distros that do have o=rw permissions on /dev/kvm.

It does seem this issue can be worked around by using `-v` instead of `--device`.  So for `/dev/kvm` that causes everyone a lot of problems on Debian-like distros:
```
# works and maps the device node into the container
podman run --privileged --security-opt label=disable -v /dev/kvm:/dev/kvm

# doesn't work, no /dev/kvm exists in the container
podman run --privileged --security-opt label=disable --device /dev/kvm
```

**Output of `podman version`:**

```
podman version 3.4.2
```

**Output of `podman info`:**

```
host:
  arch: amd64
  buildahVersion: 1.23.1
  cgroupControllers: []
  cgroupManager: cgroupfs
  cgroupVersion: v1
  conmon:
    package: 'conmon: /usr/libexec/podman/conmon'
    path: /usr/libexec/podman/conmon
    version: 'conmon version 2.1.2, commit: '
  cpus: 12
  distribution:
    codename: focal
    distribution: ubuntu
    version: "20.04"
  eventLogger: journald
  hostname: goln-5cl17g3
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1684400513
      size: 1
    - container_id: 1
      host_id: 231072
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1684455083
      size: 1
    - container_id: 1
      host_id: 231072
      size: 65536
  kernel: 5.15.0-53-generic
  linkmode: dynamic
  logDriver: journald
  memFree: 48252440576
  memTotal: 67052859392
  ociRuntime:
    name: crun
    package: 'crun: /usr/bin/crun'
    path: /usr/bin/crun
    version: |-
      crun version UNKNOWN
      commit: ea1fe3938eefa14eb707f1d22adff4db670645d6
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +YAJL
  os: linux
  remoteSocket:
    path: /run/user/1684455083/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: true
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: false
  serviceIsRemote: false
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: 'slirp4netns: /usr/bin/slirp4netns'
    version: |-
      slirp4netns version 1.1.8
      commit: unknown
      libslirp: 4.3.1-git
      SLIRP_CONFIG_VERSION_MAX: 3
      libseccomp: 2.4.3
  swapFree: 1218957312
  swapTotal: 1218957312
  uptime: 20m 54.74s
plugins:
  log:
  - k8s-file
  - none
  - journald
  network:
  - bridge
  - macvlan
  volume:
  - local
registries:
  search:
  - docker.io
  - quay.io
store:
  configFile: ${HOME}/.config/containers/storage.conf
  containerStore:
    number: 0
    paused: 0
    running: 0
    stopped: 0
  graphDriverName: overlay
  graphOptions: {}
  graphRoot: ${HOME}/.local/share/containers/storage
  graphStatus:
    Backing Filesystem: extfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
    Using metacopy: "false"
  imageStore:
    number: 3
  runRoot: /run/user/1684455083/containers
  volumePath: ${HOME}/.local/share/containers/storage/volumes
version:
  APIVersion: 3.4.2
  Built: 0
  BuiltTime: Wed Dec 31 16:00:00 1969
  GitCommit: ""
  GoVersion: go1.15.2
  OsArch: linux/amd64
  Version: 3.4.2
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman` or `brew info podman`):**

```
$ apt list podman
Listing... Done
podman/unknown,now 100:3.4.2-5 amd64 [installed]
podman/unknown 100:3.4.2-5 arm64
podman/unknown 100:3.4.2-5 armhf
podman/unknown 100:3.4.2-5 s390x
```

**Have you tested with the latest version of Podman and have you checked [the Podman Troubleshooting Guide](https://github.com/containers/podman/blob/main/troubleshooting.md)?**

N/A, not in troubleshooting

podman 3.4.2 is the latest version available for most non-Fedora distros since there doesn't seem to be enough community support for maintaining podman in any other package repos.  There is no real method to test any newer version on any non-Fedora systems as a result since building from source has spotty success at best outside of Fedora-like distros it seems.

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Comment by rhatdan** [comment]
Works for me?

```
$ podman run -v /dev/kvm:/dev/kvm alpine ls -l /dev/kvm
crw-rw-rw-    1 nobody   nobody     10, 232 Dec  2 11:46 /dev/kvm
podman (tmpfs) $ sudo sh
Place your right index finger on the fingerprint reader
# sudo chmod 660 /dev/kvm 
$ podman run -v /dev/kvm:/dev/kvm alpine ls -l /dev/kvm
crw-rw----    1 nobody   nobody     10, 232 Dec  2 11:46 /dev/kvm
```

**Comment by rhatdan** [comment]
$ podman run --device /dev/kvm alpine ls -l /dev/kvm
crw-rw----    1 nobody   nobody     10, 232 Dec  2 11:46 /dev/kvm

**Comment by rhatdan** [comment]
Perhaps this is fixed in upstream.

**Comment by mtalexan** [comment]
> Works for me?
> 
> ```
> $ podman run -v /dev/kvm:/dev/kvm alpine ls -l /dev/kvm
> crw-rw-rw-    1 nobody   nobody     10, 232 Dec  2 11:46 /dev/kvm
> podman (tmpfs) $ sudo sh
> Place your right index finger on the fingerprint reader
> # sudo chmod 660 /dev/kvm 
> $ podman run -v /dev/kvm:/dev/kvm alpine ls -l /dev/kvm
> crw-rw----    1 nobody   nobody     10, 232 Dec  2 11:46 /dev/kvm
> ```

That is indeed the workaround to the bug. Using `-v` instead of `--device`.


> Perhaps this is fixed in upstream.

Did this work for you on Alpine? What version of Postman are you using?

**Comment by rhatdan** [comment]
I work on fedora. But I doubt podman would care about its distribution.

$ podman -v
podman version 4.3.1

**Comment by mtalexan** [comment]
> I work on fedora. But I doubt podman would care about its distribution.
> 
> $ podman -v podman version 4.3.1

Ah, I was having trouble following your logic in the first reply, mixing up what was and wasn't inside the container, but now I've got it. The distro doesn't matter, you're right.  

Thanks for the podman version info, so the bug was fixed sometime between 3.4.2 and 4.3.1.

**Comment by mtalexan** [comment]
I've updated the title for future searchers to find. Thanks
